src/libraries/System.Security.Cryptography.Encoding/tests/Cbor/CborReaderState.cs

```diff
@@ -0,0 +1,133 @@
+// See the LICENSE file in the project root for more information.
+
+#nullable enable
+
+namespace System.Formats.Cbor
+{
+    /// <summary>
+    ///   Specifies the state of a CborReader instance.
+    /// </summary>
+    public enum CborReaderState
+    {
+        /// <summary>
+        ///   Indicates the undefined state.
+        /// </summary>
+        None = 0,
+
+        /// <summary>
+        ///   Indicates that the next CBOR data item is an unsigned integer (major type 0).
+        /// </summary>
+        UnsignedInteger,
+
+        /// <summary>
+        ///   Indicates that the next CBOR data item is a negative integer (major type 1).
+        /// </summary>
+        NegativeInteger,
+
+        /// <summary>
+        ///   Indicates that the next CBOR data item is a byte string (major type 2).
+        /// </summary>
+        ByteString,
+
+        /// <summary>
+        ///   Indicates that the next CBOR data item denotes the start of an indefinite-length byte string (major type 2).
+        /// </summary>
+        StartByteString,
+
+        /// <summary>
+        ///   Indicates that the <see cref="CborReader"/> is at the end of an indefinite-length byte string (major type 2).
+        /// </summary>
+        EndByteString,
+
+        /// <summary>
+        ///   Indicates that the next CBOR data item is a UTF-8 string (major type 3).
+        /// </summary>
+        TextString,
+
+        /// <summary>
+        ///   Indicates that the next CBOR data item denotes the start of an indefinite-length UTF-8 text string (major type 3).
+        /// </summary>
+        StartTextString,
+
+        /// <summary>
+        ///   Indicates that the <see cref="CborReader"/> is at the end of an indefinite-length UTF-8 text string (major type 3).
+        /// </summary>
+        EndTextString,
+
+        /// <summary>
+        ///   Indicates that the next CBOR data item denotes the start of an array (major type 4).
+        /// </summary>
+        StartArray,
+
+        /// <summary>
+        ///   Indicates that the <see cref="CborReader"/> is at the end of an array (major type 4).
+        /// </summary>
+        EndArray,
+
+        /// <summary>
+        ///   Indicates that the next CBOR data item denotes the start of a map (major type 5).
+        /// </summary>
+        StartMap,
+
+        /// <summary>
+        ///   Indicates that the <see cref="CborReader"/> is at the end of a map (major type 5).
+        /// </summary>
+        EndMap,
+
+        /// <summary>
+        ///   Indicates that the next CBOR data item is a semantic tag (major type 6).
+        /// </summary>
+        Tag,
+
+        /// <summary>
+        ///   Indicates that the next CBOR data item is a simple value (major type 7).
+        /// </summary>
+        SimpleValue,
+
+        /// <summary>
+        ///   Indicates that the next CBOR data item is an IEEE 754 Half-Precision float (major type 7).
```

Comment:
Since you're pointing out all of the major types, does it make sense for the simple values to call out their simple value type? e.g.

```suggestion
        ///   Indicates that the next CBOR data item is an IEEE 754 Half-Precision float (major type 7, simple value type 25).
```

Reply:
Technically speaking, float encodings are not considered simple values even though they share the same major type. Simple value 25 (encoded as `f819`) is different than a half-precision encoding (e.g. `f97e00`). To avoid confusion use of simple values 24-31 is discouraged by the spec, and the current implementation will throw when they are encountered (To consider: should Lax conformance maybe tolerate them?)